api/client/client.go

```diff
@@ -141,97 +207,100 @@ func collectionConfigToPb(c db.CollectionConfig) (*pb.CollectionConfig, error) {
 }
 
 // GetDBInfo retrives db addresses and keys.
-func (c *Client) GetDBInfo(ctx context.Context, creds thread.Credentials) (*pb.GetDBInfoReply, error) {
-	signed, err := signCreds(creds)
-	if err != nil {
-		return nil, err
+func (c *Client) GetDBInfo(ctx context.Context, dbID thread.ID, opts ...db.ManagedDBOption) (*pb.GetDBInfoReply, error) {
+	args := &db.ManagedDBOptions{}
+	for _, opt := range opts {
+		opt(args)
 	}
+	ctx = thread.NewTokenContext(ctx, args.Token)
 	return c.c.GetDBInfo(ctx, &pb.GetDBInfoRequest{
-		Credentials: signed,
+		DbID: dbID.Bytes(),
 	})
 }
 
-// NewCollection creates a new collection
-func (c *Client) NewCollection(ctx context.Context, creds thread.Credentials, config db.CollectionConfig) error {
-	signed, err := signCreds(creds)
-	if err != nil {
-		return err
+// NewCollection creates a new collection.
+// @todo: This should take some thread auth, but collections currently do not involve a thread.
+func (c *Client) NewCollection(ctx context.Context, dbID thread.ID, config db.CollectionConfig, opts ...db.ManagedDBOption) error {
+	args := &db.ManagedDBOptions{}
+	for _, opt := range opts {
+		opt(args)
 	}
 	cc, err := collectionConfigToPb(config)
 	if err != nil {
 		return err
 	}
 	_, err = c.c.NewCollection(ctx, &pb.NewCollectionRequest{
-		Credentials: signed,
-		Config:      cc,
+		DbID:   dbID.Bytes(),
+		Config: cc,
 	})
 	return err
 }
 
-// Create creates new instances of objects
-func (c *Client) Create(ctx context.Context, creds thread.Credentials, collectionName string, items ...interface{}) ([]string, error) {
-	signed, err := signCreds(creds)
-	if err != nil {
-		return nil, err
+// Create creates new instances of objects.
+func (c *Client) Create(ctx context.Context, dbID thread.ID, collectionName string, instances Instances, opts ...db.TxnOption) ([]string, error) {
```

Comment:
Oh, good question. I like the variadic style more. It's just that on the server is no longer possible in a clean way. So, toss up: Is it more important to have the APIs match, or to have a simpler gRPC client?

Reply:
But this isn't varadic style... `Instances` is just []interface{}`. The varadic style also isn't possible here because we're using `TxnOption` as the varadic last parameter. So if I read this right, what we have on the remote is (using `Create` as an example):

`CreateMany(list)`
`CreateOne(single)`

and on the client

`Create(list)`

It makes sense that `Create(list)` is the most general, and we could actually just do that on the remote as well and get rid of the single and many flavors. Just to clarify, varadic instances are not used anywhere anymore.

But since you introduced the single and many flavors, I actually like that because it makes the api easier to use in the most common use case of creating one new object at a time. 

I'd vote for consistency between the remote and client, and I think I'd say that your single and many flavors are a nice way to go.

Reply:
Oh, right, of course... Okay, sounds good. I'm gonna kick the can on updating the client in this PR, but I can create an issue.

Reply:
Sounds good